volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -1581,3 +1587,77 @@ def has_capability(self, capability: str) -> bool:
 
         cap_value = 1 << CAPABILITIES.index(capability)
         return cap_value & self.get_capabilities() != 0
+
+
+class net_device(objects.StructType):
+    def get_hardware_address(self):
+        parent_layer = self._context.layers[self.vol.layer_name]
+        if self.has_member("perm_addr"):
+            raw_addr = self.perm_addr[0 : self.addr_len]
+        else:  # perm_addr is not found in older kernels
+            raw_addr = parent_layer.read(self.dev_addr, self.addr_len, pad=True)
```

Comment:
Liiiiiittle bit ugly, but it's only a line, so I guess that'll be ok.  5:S  Can't think of how to improve it, so....